### PR TITLE
@types/parse : add "Parse.User.currentAsync()"

### PR DESCRIPTION
**Pull request by yomybaby**
https://parseplatform.org/Parse-SDK-JS/api/2.2.1/Parse.User.html#.currentAsync

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://parseplatform.org/Parse-SDK-JS/api/2.2.1/Parse.User.html#.currentAsync
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

**Review by agoldis — CHANGES_REQUESTED** [review]
Thank you, nice job here! 👍🏻
Just a small change (see the inline comment) and we are good to go.

Also, it would be nice to have a small test. e.g.:

```typescript 
async function test_user() {
    const asyncUser = await Parse.User.currentAsync();
    if (asyncUser) {
        asyncUser.set("email", "email@example.com");
    }
// ...
```

You can quickly run the test by:
`npx tsc --watch --types ./types/parse/ ./types/parse/parse-tests.ts`

and seeing to errors reported

File: types/parse/index.d.ts [hunk 1]
```diff
@@ -621,6 +622,7 @@ subscription.on('close', () => {});
     class User extends Object {
 
         static current(): User | undefined;
+        static currentAsync(): Promise<User>;
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply types/parse/index.d.ts]: Yes! You are right! I fixed it and pushed!

File: types/parse/parse-tests.ts [hunk 2]
```diff
@@ -1,124 +1,129 @@
-class GameScore extends Parse.Object {
+import { parse } from 'path';
```
Comment (agoldis) [inline types/parse/parse-tests.ts]: why is it here? didn't find any usage

**Review by agoldis — CHANGES_REQUESTED** [review]
@yomybaby We are almost there! 
Please revert the styling to the original one, it makes it hard to track the changes, also seems like there's an unused import in the test file.

File: types/parse/parse-tests.ts (hunk 2 above)
Comment (yomybaby) [reply types/parse/parse-tests.ts]: Oh it's accident. Sometimes "auto import" feature is annoying. Thanks you. 🙏

**Review by agoldis — APPROVED** [review]
🎉

**Pull request merged**